moonphase@techi-freki/files/moonphase@techi-freki/js/translator.js

```diff
@@ -0,0 +1,15 @@
+const { get_home_dir } = imports.gi.GLib;
+const GetText = imports.gettext;
+
+class Translator {
+    constructor(uuid) {
+        this.uuid = uuid;
+    }
+
+    translate(str) {
+        GetText.bindtextdomain(this.uuid, `${get_home_dir()}/.local/share/locale`);
+        let translated = GetText.dgettext(this.uuid, str);
+        if (translated !== str) return translated;
+        return str;
+    }
+}
```

Comment:
Using this as an internal class instead of the standard gettext invocation means that the default scripts will not pick up strings for translation properly which can pose a challenge for translators. You can see this if you run `./cinnamon-spices-makepot moonphase@techi-freki` on your system and check the output.

Reply:
Thanks Rick! I wasn't aware of that script. I have removed that class and the script looks like it worked correctly.

Reply:
Just one more quick note that you could make it global and only have to define it once. (Check other applets for an example here.) But this works and we can get this merged if you're ready.